console/v1/types_console_yaml_sample.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleYAMLSample is an extension for customizing OpenShift web console YAML samples.
+type ConsoleYAMLSample struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              ConsoleYAMLSampleSpec `json:"spec"`
+}
+
+// ConsoleYAMLSampleSpec is the desired YAML sample configuration.
+// Samples will appear with their descriptions in a samples sidebar
+// when creating a resources in the web console.
+type ConsoleYAMLSampleSpec struct {
+	// targetResource contains apiVersion and kind of the resource
+	// YAML sample is representating.
+	TargetResource metav1.TypeMeta `json:",targetResource"`
+	// title of the YAML sample.
+	// +kubebuilder:validation:Pattern=^(.|\s)*\S(.|\s)*$
```

Comment:
So unfortunately for this to work to generate the appropriate validation, the `kubebuilder` tag needs to be on a `type`.  See https://github.com/openshift/api/pull/468/files#diff-b2e8eb141280483ab2ddd2a10a65ec30R54.

Not really difficult, but a little tedious:

```golang
// +kubebuilder:validation:Pattern=^(.|\s)*\S(.|\s)*$
type ConsoleYAMLSampleDescription string
```
etc.